src/nsolid.h

```diff
@@ -1015,14 +1016,86 @@ class NODE_EXTERN Snapshot {
    */
   static int StopTrackingHeapObjectsSync(SharedEnvInst envinst);
 
+  /**
+   * @brief Will start sampling heap allocations.
+   *
+   * @param envinst SharedEnvInst of thread to take a sampled snapshot from.
+   * @param duration duration in milliseconds of the heap profiler after which
+   * the sampled heap snapshot will be returned in the callback.
+   * @param cb callback function with the following signature:
+   * `cb(int status, std::string snapshot, ...Data)`. It will be called from the
+   * NSolid thread.
+   * @param data variable number of arguments to be propagated to the callback.
+   * @return NSOLID_E_SUCCESS in case of success or a different NSOLID_E_
+   * error value otherwise.
+   */
+  template <typename Cb, typename... Data>
+  static int StartSampling(SharedEnvInst envinst,
+                           uint64_t duration,
+                           Cb&& cb,
+                           Data&&... data);
+
+  /**
+   * @brief Will start sampling heap allocations.
+   *
+   * @param envinst SharedEnvInst of thread to take the snapshot from.
+   * @param sample_interval frequency at which the heap profiler samples the
+   * heap and captures data
+   * @param stack_depth refers to the depth of the call stack that will be
+   * captured during heap profiling
+   * @param flags additional flags or options that can be used when
+   * starting heap profiling
```

Comment:
Nit: name the units used for `sample_interval` (I think they're bytes). Also add a link to v8docs for the flags enum